src/MySqlConnector/MySqlClient/MySqlDataReader.cs

```diff
@@ -191,6 +193,22 @@ public override void Close()
 		}
 #endif
 
+		public override T GetFieldValue<T>(int ordinal)
+		{
+			if (typeof(T) == typeof(DateTimeOffset))
+				return (T) Convert.ChangeType(GetDateTimeOffset(ordinal), typeof(T));
```

Comment:
Isn't `(T) Convert.ChangeType` redundant here because we know `T` is `DateTimeOffset`?